src/components/badge/index.js

```diff
@@ -97,6 +97,17 @@ export default class Badge extends PureBaseComponent {
     }
   }
 
+  getAccessibilityProps() {
+    const {onPress, icon, label} = this.props;
+    
+    return {
+    accessibilityLabel: icon ? 'badge' : `badge ${label}` || 'badge',
```

Comment:
what is this
```
`badge ${label}` || 'badge'
```
you're using pipe for a fallback value?

Reply:
also consider using `imagebutton` role for when the badge has onPress with an icon

Reply:
Fixed.
Regarding `imagebutton`, it can have onPress in all badge variation (icon, text, no text - just bg color).